src/request_parser.ts

```diff
@@ -37,15 +46,43 @@ export const parseTakerRequest = (req: Pick<express.Request, 'headers' | 'query'
 
         let takerRequest: TakerRequest;
         if (query.sellAmountBaseUnits !== undefined && query.buyAmountBaseUnits === undefined) {
-            takerRequest = {
-                ...takerRequestBase,
-                sellAmountBaseUnits: new BigNumber(query.sellAmountBaseUnits),
-            };
+            if (protocolVersion === '3') {
+                takerRequest = {
+                    ...takerRequestBase,
+                    protocolVersion,
+                    sellAmountBaseUnits: new BigNumber(query.sellAmountBaseUnits),
+                };
+            } else {
+                if (query.txOrigin !== undefined) {
+                    takerRequest = {
+                        ...takerRequestBase,
+                        sellAmountBaseUnits: new BigNumber(query.sellAmountBaseUnits),
+                        protocolVersion,
+                        txOrigin: query.txOrigin,
+                    };
+                } else {
+                    throw new Error('v4 request must specify txOrigin');
```

Comment:
Since this is a little duplicated code, I'm going to go ahead and refactor this check in it's own statement